### PR TITLE
SPI peripheral 2.0

### PR DESCRIPTION
**Pull request by phil-levis**
This updates the SPI peripheral to use the Tock 2.0 syscall API.

I managed to get a board to loopback from a USART SPI controller to a SPI peripheral. The system calls work correctly.

### CODE REVIEW

File: libtock/ambient_light.c [hunk 1]
```diff
@@ -38,20 +38,10 @@ int ambient_light_read_intensity_sync(int* lux_value) {
...
 }
 
 int ambient_light_start_intensity_reading(void) {
-  syscall_return_t ret = command2(DRIVER_NUM_AMBIENT_LIGHT, 1, 0, 0);
-  if (ret.type == TOCK_SYSCALL_SUCCESS) {
-    return TOCK_SUCCESS;
-  } else {
-    return tock_error_to_rcode(ret.data[0]);
-  }
+  return command(DRIVER_NUM_AMBIENT_LIGHT, 1, 0, 0);
```
Comment (hudson-ayers) [inline libtock/ambient_light.c:45]: This seems to revert the ambient light --> 2.0 changes, and should probably be removed?

File: libtock/spi_peripheral.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "tock.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define SPI_SLAVE 0x20002
```
Comment (hudson-ayers) [inline libtock/spi_peripheral.h:9]:
```suggestion
#define SPI_PERIPHERAL 0x20002
```

**Review by hudson-ayers — CHANGES_REQUESTED** [review]

File: libtock/ambient_light.c (hunk 1 above)
Comment (phil-levis) [reply libtock/ambient_light.c:45]: Hm, yeah. I suspect this is an uncrustify thing.

**Comment by phil-levis** [comment]
I'm leaning towards closing this PR and just yanking `spi_peripheral`; going to talk with Tyler and Michaela first, though.

**Comment by phil-levis** [comment]
@hudson-ayers I managed to test this (successfully) and addressed your comments. Could you take a second look.

**Review by hudson-ayers — APPROVED** [review]
LGTM

**Pull request merged**